rust/src/mqtt/mqtt.rs

```diff
@@ -492,7 +518,11 @@ impl MQTTState {
                 Ok((rem, msg)) => {
                     SCLogDebug!("response msg {:?}", msg);
                     if let MQTTOperation::TRUNCATED(ref trunc) = msg.op {
-                        SCLogDebug!("found truncated with skipped {} current len {}", trunc.skipped_length, current.len());
+                        SCLogDebug!(
+                            "found truncated with skipped {} current len {}",
+                            trunc.skipped_length,
+                            current.len()
+                        );
                         if trunc.skipped_length >= current.len() {
                             skipped = true;
                             self.skip_response = trunc.skipped_length - current.len();
```

Comment:
@satta there is a bug here : if we get in the case just below `current = &current[trunc.skipped_length..];` we should not run `consumed += current.len() - rem.len();` (because we have changed current and the integer may underflow)

Reply:
@catenacyber should this issue block merge of this PR? Or is the bug outside of the PR?

Reply:
Outside this PR
But I do not want to create another PR while this one is not merged (so that I do not have to deal with conflicts) ;-)